commit 939e13c29fc6a7f5355d1d6e616a60222affb284
Author: Thomas J. Moore <darktjm@gmail.com>
Date:   Sun Apr 19 22:11:38 2020 -0500

    Only write changelog if changed

diff --git a/src/downloader.cpp b/src/downloader.cpp
index 6968ab8..3efea53 100644
--- a/src/downloader.cpp
+++ b/src/downloader.cpp
@@ -2331,6 +2331,13 @@ void Downloader::saveChangelog(const std::string& changelog, const std::string&
         }
     }
 
+    try {
+	std::string olog;
+	boost::filesystem::load_string_file(filepath, olog);
+	if (olog == changelog)
+	    return;
+    }
+    catch (std::ios_base::failure& ex) {} // ignore missing file
     std::ofstream ofs(filepath);
     if (ofs)
     {

