commit 50b41cad7ff37a78604067776bafb918df125de5
Author: Thomas J. Moore <darktjm@gmail.com>
Date:   Sun Nov 15 19:43:30 2020 -0600

    Suppress SIGPIPE during cache update

diff --git a/main.cpp b/main.cpp
index 78be7a5..6ec0064 100644
--- a/main.cpp
+++ b/main.cpp
@@ -15,6 +15,7 @@
 #include <fstream>
 #include <boost/filesystem.hpp>
 #include <boost/program_options.hpp>
+#include <signal.h>
 
 namespace bpo = boost::program_options;
 Config Globals::globalConfig;
@@ -26,6 +27,7 @@ template<typename T> void set_vm_value(std::map<std::string, bpo::variable_value
 
 int main(int argc, char *argv[])
 {
+    signal(SIGPIPE, SIG_IGN);
     rhash_library_init();
 
     // Constants for option selection with include/exclude

